### PR TITLE
Make KeyframesPercentage::parse faster

### LINKED ISSUE #20179: Make KeyframesPercentage::parse faster.
Right now it does:

https://github.com/servo/servo/blob/0eb8b1f4c0736ec25dc391d119c03d1b49644d81/components/style/stylesheets/keyframes_rule.rs#L128-L144

That's more inefficient that what it needs to be, since it really only needs to get the next token once using `next()`, and then match on `Token::Ident` and `Token::Percentage`.

No new tests need to be added for this refactoring.

### PR DESCRIPTION
**Pull request by Beta-Alf**
Minor refactoring of the parsing of keyframes.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix  #20179 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because it is a minor refactor so tests are already in place. The original Issue explicitly says so.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20185)

### CODE REVIEW

File: components/style/stylesheets/keyframes_rule.rs
Comment (emilio) [inline components/style/stylesheets/keyframes_rule.rs]: nit: indentation here looks wrong. Also, there's no need to clone the token except in the last branch, which can be:

```rust
ref token => Err(input.new_unexpected_token_error(token.clone()),
```

File: components/style/stylesheets/keyframes_rule.rs
Comment (emilio) [inline components/style/stylesheets/keyframes_rule.rs]: nit: Maybe use braces here:

```rust
Token::Ident(ref identifier) if identifier.as_ref().eq_ignore_ascii_case("from") => {
    Ok(KeyframePercentage::new(0.))
}
```

And same below. It's a bit more verbose but easier to read I think. Your call anyway :)

**Review by emilio — APPROVED** [review]
Looks great with those fixed. Thanks!

**Comment by emilio** [comment]
Also, please squash your commits before landing if you don't mind? Thanks a lot again for fixing this!

**Comment by Beta-Alf** [comment]
Thank you for the new feedback, I have two questions:
1. Pushing the clone down does not work so easy because then `token` will create a mutable borrow on input and I reborrow it when creating the error in the default case. My question: what is the rustic solution to this?
2. As to the squash: Should I just create a new "squashed" branch and open another PR referencing this one? Because history rewriting feels kind of dirty.

**Comment by emilio** [comment]
I see, this looks good to me. Regarding squashing, I'd just do it in this same branch, rewriting history is fine :)

**Comment by Beta-Alf** [comment]
So it seems like some snafu occurs in CI: https://taskcluster-artifacts.net/PoaAmKsMQbS3Z5hwC6gtYg/0/public/logs/live_backing.log
Tests all (unit, tidy) pass for me.

**Comment by emilio** [comment]
Oh the taskcluster stuff doesn't block landing, no worries. It is known, see: https://groups.google.com/forum/#!topic/mozilla.dev.servo/lttaOHEMzes

**Review by emilio — APPROVED** [review]
Looks great, thank you :)

**Comment by tigercosmos** [comment]
@Beta-Alf I found your commit's username and email are not as same as your github's. So Github does not know you are contributor in this situation.  Just a suggestion.

**Pull request merged**
